src/cloudsc_mix/cloudsc_c_k_caching.cpp

```diff
@@ -0,0 +1,2846 @@
+/*
+ * (C) Copyright 1988- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation
+ * nor does it submit to any jurisdiction.
+ */
+
+#if GPU_LANG == CUDA_LANG
+#include "cuda.h"
+#elif GPU_LANG == HIP_LANG
+#include "hip/hip_runtime.h"
+#elif GPU_LANG == SYCL_LANG
+#include <CL/sycl.hpp>
+#endif
```

Comment:
Not necessarily required but just to point out that we could use here the same pattern that ectrans is using to switch between HIP and CUDA implementations via header files that rewrite relevant API calls on demand:
https://github.com/ecmwf-ifs/ectrans/blob/main/src/trans/gpu/algor/hicblas.h
With specific overloads for [HIP](https://github.com/ecmwf-ifs/ectrans/blob/main/src/trans/gpu/algor/hicblas_hip.h) and [CUDA](https://github.com/ecmwf-ifs/ectrans/blob/main/src/trans/gpu/algor/hicblas_cuda.h)